core/src/main/java/org/elasticsearch/action/admin/cluster/stats/ClusterStatsNodes.java

```diff
@@ -560,12 +571,16 @@ private NetworkTypes(final List<NodeInfo> nodeInfos) {
         public XContentBuilder toXContent(final XContentBuilder builder, final Params params) throws IOException {
             builder.startObject("transport_types");
             for (final Map.Entry<String, AtomicInteger> entry : transportTypes.entrySet()) {
-                builder.field(entry.getKey(), entry.getValue().get());
+                if (Strings.hasText(entry.getKey())) {
```

Comment:
should the check be done here, or should such an element never get added to the list at the first place?